tests/build/build_templates.go

```diff
@@ -797,3 +803,46 @@ func applyAdditionalTagsInPipelineBundle(customDockerBuildBundle string, additio
 	}
 	return newDockerBuildPipeline.String(), nil
 }
+
+func ensureOriginalDockerfileIsPushed(hub *framework.ControllerHub, pr *tektonpipeline.PipelineRun) {
+	binaryImage := build.GetBinaryImage(pr)
+	Expect(binaryImage).ShouldNot(BeEmpty())
+
+	binaryImageRef, err := reference.Parse(binaryImage)
+	Expect(err).Should(Succeed())
```

Comment:
nit: would `Expect(err).ShouldNot(HaveOccurred())` be better? 
It's a pattern mentioned in the [gomega docs](https://onsi.github.io/gomega/#handling-errors)
The docs say that `.Should(Succeed)` is intended to be used with functions which have error as first/only return value. https://onsi.github.io/gomega/#succeed

Reply:
I'm not an expert of Gomega to be honest :) After reading the document you give, I think this statement:

> You should not use a function with multiple return values (like DoSomethingHard) with Succeed

means this form should not be used:

```Except(reference.Parse(binaryImage)).Should(Succeed())```

However, the line `Expect(err).Should(Succeed())` just passes a single value to `Except`.

Or, do I miss the content you read?

Anyway, I would keep this line unchanged for now because the CI execution is expensive. If such style of Gomega calls really have to be changed, we can open a separate PR for it specifically.

Reply:
Sure, it works. It's just a stylistic choice.